### DISCUSSION TITLE
Why is this not remote caching

### DISCUSSION THREAD

**Discussion by mike-hogan** [vercel/turborepo#2224] [Help] [answered]
I have remote caching enabled.  When I do --dry-run=json, I get this output for a task that is not remote caching:

```
{
  "taskId": "@docsndata/google-docs2#build",
  "task": "build",
  "package": "@docsndata/google-docs2",
  "hash": "f01e22429c9263e4",
  "cacheState": {
    "local": true,
    "remote": false
  },
  "command": "tsc -p tsconfig.json \u0026\u0026 tsc -p tsconfig-cjs.json",
  "outputs": [
    ".svelte-kit/**",
    "dist/**",
    "package/**"
  ],
  "excludedOutputs": null,
  "logFile": "google/docs2/.turbo/turbo-build.log",
  "directory": "google/docs2",
  "dependencies": [
    "@docsndata/model-types#build",
    "@docsndata/model-core2#build",
    "@docsndata/lang-lodash#build",
    "@docsndata/model-team#build",
    "@docsndata/model-testing2#build",
    "@docsndata/ops-monitor#build",
    "@docsndata/lang-assertions#build",
    "@docsndata/model-docs#build",
    "@docsndata/lang-util#build",
    "@docsndata/templating-hydration#build",
    "@docsndata/templating-types#build",
    "@docsndata/google-api#build",
    "@docsndata/lang-time#build"
  ],
  "dependents": [
    "@docsndata/apps-3webapp#build",
    "@docsndata/apps-services#build",
    "@docsndata/components-google-doc-generation#build",
    "@docsndata/google-workflows#build"
  ]
},
```

When I run the task again, I get the same hash, but `remote` stays `false`.

Other packages have `remote` as `true`, but no matter what I do, this one will not remote cache.  Any idea why?

**Comment by nathanhammond** [comment, marked as answer]
This is because you have a local copy of this cache! If you remove the directory at node_modules/.cache/turbo/f01e22429c9263e4 that will switch the state.

**Reply by mike-hogan** [reply]
I see, thx! What is the interplay between local and remote caches?  Is remote checked only if local misses?

If I want building locally to prime the remote cache (to make the CI build run faster), how can I do that?  Should I be using --remote-only?

**Reply by mike-hogan** [reply]
Actually that is a bit of a silly question, why would it check remote if local hits.  What I really was getting it was under what circumstances a local build would write to a remote cache.  I assume never, unless --remote-only is used?

**Reply by nathanhammond** [reply]
Correct.

**Reply by jonaslindstr** [reply]
Is this really so? The docs on Remote Caching say 

> Once enabled, make some changes to a workspace you are currently caching and run tasks against it with turbo run. Your cache artifacts will now be stored locally and in your Remote Cache.


i.e. once remote caching is enabled.
